File: backend/src/warehouse/domain/items/repository.py
"""Items domain repository."""

from uuid import UUID

from warehouse.domain.items.models import Category, Item
from warehouse.lib.base import BaseRepository


class CategoryRepository(BaseRepository[Category]):
    """Category repository."""

    model_type = Category


class ItemRepository(BaseRepository[Item]):
    """Item repository."""

    model_type = Item

    async def get_by_sku(self, sku: str, workspace_id: UUID) -> Item | None:
        """Get item by SKU within a workspace."""
        return await self.get_one_or_none(sku=sku, workspace_id=workspace_id)

File: backend/internal/domain/warehouse/wishlist/repository.go
package wishlist

import (
	"context"

	"github.com/google/uuid"

	"github.com/antti/home-warehouse/go-backend/internal/shared"
)

// Repository defines the interface for wishlist item persistence.
type Repository interface {
	// Save creates or updates a wishlist item.
	Save(ctx context.Context, item *Item) error

	// FindByID retrieves a wishlist item by ID within a workspace.
	FindByID(ctx context.Context, id, workspaceID uuid.UUID) (*Item, error)

	// FindByWorkspace retrieves wishlist items for a workspace with
	// pagination, optionally filtered by status (nil = all statuses).
	// Results are sorted by priority (1 = highest first), newest first
	// within a priority. Returns the items, total count, and any error.
	FindByWorkspace(ctx context.Context, workspaceID uuid.UUID, status *Status, pagination shared.Pagination) ([]*Item, int, error)

	// Delete removes a wishlist item by ID.
	Delete(ctx context.Context, id, workspaceID uuid.UUID) error
}
